Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -161,10 +161,14 @@ public class SchemeGenerator {
             return XCScheme.ExecutionAction(scriptText: action.script, title: action.name, environmentBuildable: environmentBuildable)
         }
 
-        let schemeTarget: Target?
-
-        if let targetName = scheme.run?.executable {
-            schemeTarget = project.getTarget(targetName)
+        var schemeTarget: Target? = nil
+        var pathRunnable: XCScheme.PathRunnable? = nil
+        
+        if let executable = scheme.run?.executable {
+            schemeTarget = project.getTarget(executable)
+        } else if let filePath = scheme.run?.filePath {
+            // Only use pathRunnable if no executable is provided.
+            pathRunnable = XCScheme.PathRunnable(filePath: filePath)
```

Comment:
According to the ProjSpec you wrote, it looks filePath is only available for the absolute path. 
How about throwing SchemeGenerationError if the path is not the absolute path?

You can check it by using `path.isAbsolute` in `Path` class.

Reply:
Good idea. Added.